### DISCUSSION TITLE
German model tts_models/de/thorsten/vits cannot read 4th and 5th properly

### DISCUSSION THREAD

**Discussion by OWKenobi** [coqui-ai/TTS#2020] [General Q&A]
### Describe the bug

**Problem**
A sentence like "This is my 4th attempt" reads in German "Das ist mein 4. Versuch". 

Now, "4." is pronounced as "4ter", but the AI treats the dot as a full stop, which makes it sound like "This is my four. Attempt."

German pronunciation is quite straight-forward:

1. = erster
2. = 2ter
3. = dritter
4. = 4ter
5. = 5ter
6. = 6ter
7. = 7ter
8. = 8ter
9. =9ter
10. =10ter
11. =11ter
and so on for all numbers (just adding "ter")

**Solution:**
can you add a number_norm for German? I can proof-hear it if necessary.

### To Reproduce

tts --text "Das ist mein 4. Versuch" --model_name "tts_models/de/thorsten/vits" --out_path temp.wav
play temp.wav

### Expected behavior

Pronounciation: "Das ist mein 4ter Versuch."

### Environment

```shell
standard installation (Linux Mint), latest release (git pull from 19th of September)
```

**Comment by thorstenMueller** [comment]
Hi @OWKenobi ,
using the `german_transliterate` (https://github.com/repodiac/german_transliterate) repo by @repodiac in preprocessing might help.

Calling `python3 ./core.py "Das ist mein 4. Versuch"` will bring `das ist mein vierte versuch`. Okay, it's not perfect, sounding a little bit "Hessisch" 😆 but might be worth taking a look.

**Comment by lexkoro** [comment]
Without knowing the context of the sentence this is not as straight-forward as one might think, as thorsten's example also shows.

`Das ist mein 4. Problem.` -> `Das ist mein vier(tes) Problem.`

`Das ist meine 4. Runde.` -> `Das ist meine vier(te) Runde.`

`Wir sind zu 4. unterwegs.` - `Wir sind zu vier(t) unterwegs.`

**Comment by OWKenobi** [comment]
Hello lexkoro,

you are right, some context is indeed needed - I didn't think about that. Although I think the "Hessisch" variant should be default at least, since it is way better than the current behaviour.

the idea of some manual preprocessing is a good one. It could be as easy as string.replace some fixed things, so I guess before I install a not-perfect-working 3rd party unmaintained code, I'll quickly glue it together myself.

I wonder if someone actually made the effort of training a small neural net on this task. Could be trained on real text that got read by people, with their audio being stt-ed back to text where all abbrevations have been solved. This neural net wouldn't even have to be that big, cause it only has to detect some basic grammar rules and a few exceptions.

**Comment by erogol** [comment]
I move it to the discussion as it is not a software issue.
